doc/content/design/sm-supported-image-formats.md

```diff
@@ -65,12 +49,64 @@ required-cluster-stack ( RO)       :
 supported-image-formats ( RO)       : vhd, raw, qcow2
 ```
 
-This change impacts the SM data model, and as such, the XAPI database version will
-be incremented.
+## Implementation details
+
+The `supported_image_formats` field will be populated by retrieving information
+from the SMAPI drivers. Specifically, each driver will update its `DRIVER_INFO`
+dictionary with a new key, `supported_image_formats`, which will contain a list
+of strings representing the supported image formats
+(for example: `["vhd", "raw", "qcow2"]`).
+
+### Driver behavior without `supported_image_formats`
+
+If a driver does not provide this information (as is currently the case with
+existing drivers), the default value will be an empty array (`[]`). This signifies
+that the driver determines which format to use when creating VDI. During a migration,
+the destination driver will choose the format of the VDI if none is explicitly
+specified. This ensures backward compatibility with both current and future drivers.
+
+### Specifying image formats for VDIs creation
+
+If the supported image format is exposed to the client, then when creating new VDI,
+user can specify the desired format via the `sm_config` parameter `type=qcow2` (or
+any format that is supported). If no format is specified, the driver will use its
```

Comment:
When the format is not supported, an error should be returned, with the supported formats (or mention that the sr can't be told about the type)

Reply:
Yes right. I will update this part.